classes/System.php

```diff
@@ -660,7 +661,15 @@ public function time_remaining($timestamp): string {
     public function log($type, $title, $contents) {
         $type = $this->clean($type);
         $title = $this->clean($title);
-        $contents = $this->clean($contents);
+
+        if (is_array($contents))
+        {
+            $contents = json_encode($contents);
+        }
+        else
+        {
+            $contents = $this->clean($contents);
+        }
```

Comment:
json_encode won't strip out single quotes that are used for SQL injection, so we still need to do that since we have no guarantees about the data passed into `log` (could include user input)

```suggestion
        if (is_array($contents))
        {
            $contents = json_encode($contents);
        }

       $contents = $this->clean($contents);
```

Reply:
I definitely understand the concern there. I can just manually log the output of the clean clan function since I added that piece in there since I was using it to log the return but running clean on it makes it hard to read or search against if you needed to query the log. 
![image](https://user-images.githubusercontent.com/10229531/188171366-e271bcf6-840e-4a47-9714-d59a1e0a6a4f.png)